src/Brahma.FSharp.OpenCL.Workflow/Workflow.fs

```diff
@@ -0,0 +1,17 @@
+module Brahma.FSharp.OpenCL.Workflow.Workflow
+
+open Brahma.OpenCL
+open Brahma.FSharp.OpenCL.Workflow.BasicMonads.Reader
+
+type OpenCLContext(provider: ComputeProvider) =
```

Comment:
Can we add a constructor of OpenCLContext which takes not provider, but a device or even platform name?

Can we add a constructor without arguments, which tries to create a provider for the first openCL-compatible device? It may be useful for tests, for example.

Reply:
I have added one constructor with optional parameters (platform name and device type) that covered almost all your suggestions. 
Also, to support proper usage of the creation `ComputeProvider` I have changed its `Create` function in the following way. Previously, this function found the first successful match with the regular expression of the platform name, and in the first platform found, it searched for a suitable device. In this case, we might not find a suitable device. For example, let's consider some intel platform with only one CPU device and some Nvidia platform with only one GPU device. Then `ComputeProvider.Create("*", DeviceType.CPU)` can raise an exception, if firstly we will try to find CPU device on Nvidia platform.